cmd/virt-handler/virt-handler.go

```diff
@@ -73,7 +74,13 @@ func main() {
 	broadcaster.StartRecordingToSink(&kubecorev1.EventSinkImpl{Interface: coreClient.Events(api.NamespaceDefault)})
 	recorder := broadcaster.NewRecorder(kubev1.EventSource{Component: "virt-handler", Host: *host})
 
-	domainManager, err := virtwrap.NewLibvirtDomainManager(domainConn, recorder)
+	var domainManager virtwrap.DomainManager
+	switch *driver {
+	case "ovirt":
+		domainManager, err = virtwrap.NewOvirtDomainManager(domainConn, recorder)
+	default:
+		domainManager, err = virtwrap.NewLibvirtDomainManager(domainConn, recorder)
+	}
```

Comment:
It is fully to be expected that oVirt will require architectural changes to utilize KubeVirt. Having VDSM plugin to KubeVirt internals is not something we will ever support. The architecture is exclusively designed to manage KVM via libvirt and nothing else. Adding pluggable virtual handler backends would significantly increase the complexity, and place an ongoing burden on development and testing alike. If it is not possible for OVirt to do what it needs via the k8s public API, then we need to do work to ensure that is possible, since enabling integration with applications via the public API is one of key goals to be met.